### DISCUSSION TITLE
TIME & FREQUENCY DOMAIN OF TRANSMITTED & RECIEVED SIGNAL

### DISCUSSION THREAD

**Discussion by CoinLAb10** [ggerganov/ggwave#59] [Q&A]
Dear Georgi

Your work is amazing, i fallowed your guidelines on how to build and test this library it worked.

i was curious on how to save the received waveform/signal so that one can analyze and draw the Time domain of the signal which can help to see the effect of noise when compared to the transmitted signal.

i have seen that you do have the section that save transmitted signal as .wav.

thank you

**Comment by ayzic45** [comment]
This should be done, by playing with the SDL library, to save the waveform of the signal being encoded or decoded. and PHP makes it easier, you can actually export the recorded/captured data before doing anything with it. then use MATLAB analyzing and graph plots(Time & Frequency Domain),  a piece of code from MATLAB forum can help plot the graph in MATLAB.

 all be best.

**Reply by ggerganov** [reply]
That is correct. The received waveform/signal is obtained directly from SDL or the respective audio backend that you use to capture the data. `ggwave` is not needed in order to plot the Time domain data of the recorded audio - you simple need to process the RAW captured samples.

In case you want to plot the transmitted audio waveform, then you can obtain it via `GGWave::takeTxAmplitudeI16()` method after calling the `GGWave::encode()` method. See the `Waver` app for example:

https://github.com/ggerganov/ggwave/blob/43bf2f17e1c8d17be979d4e489588baaecbd410a/examples/waver/common.cpp#L703-L706

And plotting:

https://github.com/ggerganov/ggwave/blob/43bf2f17e1c8d17be979d4e489588baaecbd410a/examples/waver/common.cpp#L1478-L1532
